### PR TITLE
Remove code that has been deprecated in 6.16 and marked for removal in 6.18

### PR DESCRIPTION
**Pull request by amadio**

### CODE REVIEW

**Comment by amadio** [comment]
Note: This will likely fail since some of the options being removed are still enabled in Jenkins builds (so ROOT will fail to configure). Once Jenkins is adjusted, this PR should pass.

**Comment by linev** [comment]
@amadio @Axel-Naumann 

Why you decide to remove ODBC? It is same kind of API as MySQL or Oracle.
Probably, there are not that much users, but still. 
Last relevant post on forum is from 2017: https://root-forum.cern.ch/t/26631

**Comment by Axel-Naumann** [comment]
Thanks for your comment, Sergey - this is exactly what we were after and why we do it already now for 6.18!

We are not aware of current users of the ODBC interfaces. The post you mentioned seems more of a test balloon - it would be interesting to hear whether it's actually used. We can either "undeprecate" it and add tests (!) or remove it - it has been deprecated for one release now. And IIRC even *building* the ODBC part is becoming increasingly tricky with current distros...

**Comment by linev** [comment]
> And IIRC even building the ODBC part is becoming increasingly tricky with current distros...

Just set default OFF for ODBC building. 

> We are not aware of current users of the ODBC interfaces.

I have no idea. I personally did not try it for more than 10 years.

**Comment by amadio** [comment]
Hi @linev, I think @Axel-Naumann already said everything. I just removed the things that were deprecated. We can "undeprecate" and drop the commit from this PR if needed.

**Comment by amadio** [comment]
> And IIRC even building the ODBC part is becoming increasingly tricky with current distros...

This is not true. I've always kept `-Dodbc=ON` in my builds.

**Comment by amadio** [comment]
@phsft-bot build

**Comment by amadio** [comment]
Supported options were computed before configuring, so removed options ended up enabled. Let's see if my last commit to `rootspi` fixed this.

**Comment by amadio** [comment]
@gganis Hi Gerri, could you please review also this new pull request? In particular, removals of afdsmgr, globus, castor, rfio, and bonjour. Thank you!

**Comment by amadio** [comment]
@pcanal Could you please review the PR and state an opinion about removal of ODBC? Thanks!

**Comment by amadio** [comment]
@couet Could you please review removals of Qt and Graphviz?

**Comment by pcanal** [comment]
I am leery of removing support for ODBC as it technically allows to have access to any/most database type without having to have a specific plugin for that database.  In this one case, I would at least leave it in (deprecated mode) a while longer (at least one more release ; no every body is jumping on the latest release so it might take a while for some people to notice the deprecation .. and yes we could then bring it back but it might (or might not) then need adaptions to newer things (in particular cmake improvements) that would be done easier if it is left in).

**Comment by pcanal** [comment]
castor and rfio themselves are deprecated/on-the-way out, so fair enough to remove them.

**Comment by pcanal** [comment]
HDFS is still in use (albeit it stopped growing) and the last update to the plugin was just '18 months' ago by an external contributor (Evgeny Boger <eugenyboger@gmail.com>).   Before removing it we ought to reach you to him.

**Comment by amadio** [comment]
@pcanal. Ok, I will drop the removal of ODBC then, and wait before removing HDFS. We could leave HDFS for the next release as well.

**Comment by Axel-Naumann** [comment]
I'm contacting Evgeny for HDFS.

**Comment by gganis** [comment]
@amadio Sorry, my last comment was for afdsmgrd. Still doing the rest.

**Comment by gganis** [comment]
@amadio Bonjour ok. RFIO ok (but it includes also Rfio, aka UidGid authentication, which is a different thing; I wonder if all authentication options should not be removed as a whole).

**Comment by gganis** [comment]
@amadio Castor ok.

File: config/rootrc.in [hunk 1]
```diff
@@ -438,19 +438,16 @@ ACLiC.Linkdef:          _linkdef
 # Default login name (if not defined is taken from $(HOME)).
 #UsrPwd.Login:           qwerty
 #Krb5.Login:             qwerty@LOCAL.DOM.AIN
-#Globus.Login:           cd:~/.globus cf:usercert.pem  kf:userkey.pem  ad:/etc/grid-security/certificates
 #UidGid.Login:           qwerty
```
Comment (gganis) [inline config/rootrc.in]: References to UidGid should be removed in here too.

File: etc/hostcert.conf [hunk 2]
```diff
@@ -32,9 +32,3 @@
 ###
 ### To use host certificates from a non standard location
 #  *   /etc/grid-security/cernCA/hostcert.pem     /etc/grid-security/cernCA/hostkey.pem
```
Comment (gganis) [inline etc/hostcert.conf]: This file should be removed as a whole; it only made sense for globus

**Comment by bbockelm** [comment]
@pcanal - HDFS can go.  We still use it at most of the CMS Tier-2 sites but not through the ROOT plugin.

File: etc/hostcert.conf (hunk 2 above)
Comment (amadio) [reply etc/hostcert.conf]: Ok, will do. Thanks.

File: config/rootrc.in (hunk 1 above)
Comment (amadio) [reply config/rootrc.in]: Removed references to `UidGid` authentication.

File: etc/hostcert.conf (hunk 2 above)
Comment (amadio) [reply etc/hostcert.conf]: Done.

**Comment by amadio** [comment]
Alright, I think all review comments have been addressed. Unfortunately, Jenkins is not working due to a firewall problem, but additional changes from the previous version (which passed the checks) are small, and all tests still pass locally on my machine, so I will merge this and watch the incremental builds to see if there's anything I missed.

**Comment by amadio** [comment]
Jenkins is back, so better to let the build run before merging.

**Comment by amadio** [comment]
Travis CI breakage is not my fault, merging as Jenkins test passed.

**Pull request merged**